CMakeLists.txt

```diff
@@ -0,0 +1,157 @@
+# ----------------------------------------------------------------------
+# 2020 Bernd Pfrommer bernd.pfrommer@gmail.com
+#
+cmake_minimum_required(VERSION 3.10)
+
+if (${CMAKE_VERSION} VERSION_LESS 3.15)
+   project(vilib VERSION 1.0.0 DESCRIPTION "fast cuda fast feature tracking" LANGUAGES CXX)
+else()
+   project(vilib VERSION 1.0.0 DESCRIPTION "fast cuda fast feature tracking" LANGUAGES CXX CUDA)
+endif()
+include (CMakePrintHelpers)
+include (CMakePackageConfigHelpers)
+include (CPack)
+include (GNUInstallDirs)
+
+if (${CMAKE_VERSION} VERSION_LESS 3.15)
+   find_package(CUDA REQUIRED)
+   set(CUDA_LINK_LIBRARIES_KEYWORD PUBLIC)
+   cuda_select_nvcc_arch_flags(ARCH_FLAGS)
+   message(STATUS "Found CUDA ${CUDA_VERSION_STRING} at ${CUDA_TOOLKIT_ROOT_DIR}")
+endif()
+
+
+find_package(OpenCV REQUIRED COMPONENTS core)
+find_package(Eigen3 REQUIRED)
+
+option(BUILD_SHARED_LIBS "build shared libraries" ON)
+cmake_print_variables(BUILD_SHARED_LIBS)
+
+
+set(CUSTOM_OPENCV_PATH "" CACHE PATH "path to custom opencv version")
+
+add_library(vilib SHARED
+  visual_lib/src/cuda_common.cpp
+  visual_lib/src/timer.cpp
+  visual_lib/src/common/framebundle.cpp
+  visual_lib/src/common/frame.cpp
+  visual_lib/src/common/point.cpp
+  visual_lib/src/feature_tracker/feature_tracker_base.cpp
+  visual_lib/src/feature_tracker/feature_tracker_gpu.cpp
+  visual_lib/src/storage/ros.cpp
+  visual_lib/src/storage/opencv.cpp
+  visual_lib/src/storage/pyramid_pool.cpp
+  visual_lib/src/storage/subframe.cpp
+  visual_lib/src/storage/subframe_pool.cpp
+  visual_lib/src/preprocess/pyramid_cpu.cpp
+  visual_lib/src/preprocess/image_preprocessing.cpp
+  visual_lib/src/statistics.cpp
+  visual_lib/src/feature_detection/fast/opencv/fast_cpu.cpp
+  visual_lib/src/feature_detection/fast/rosten/fast_cpu.cpp
+  visual_lib/src/feature_detection/fast/rosten/fast_9.cpp
+  visual_lib/src/feature_detection/fast/rosten/fast_12.cpp
+  visual_lib/src/feature_detection/fast/rosten/fast_10.cpp
+  visual_lib/src/feature_detection/fast/rosten/nonmax.cpp
+  visual_lib/src/feature_detection/fast/rosten/fast_11.cpp
+  visual_lib/src/feature_detection/fast/rosten/fast.cpp
+  visual_lib/src/feature_detection/fast/fast_gpu.cpp
+  visual_lib/src/feature_detection/detector_base_gpu.cpp
+  visual_lib/src/feature_detection/detector_benchmark.cpp
+  visual_lib/src/feature_detection/detector_base.cpp
+  visual_lib/src/timergpu.cpp
+  visual_lib/src/feature_tracker/feature_tracker_cuda_tools.cu
+  visual_lib/src/preprocess/pyramid_gpu.cu
+  visual_lib/src/feature_detection/fast/fast_gpu_cuda_tools.cu
+  visual_lib/src/feature_detection/detector_base_gpu_cuda_tools.cu
+  )
+#  visual_lib/src/cuda_common.cpp
```

Comment:
why not getting all source files by
```
file(GLOB_RECURSE SOURCES
  visual_lib/src/*.cpp
  visual_lib/src/*.cu
)
```
and probably remove the ones you don't want with
```
set(EXCLUDE_SOURCES "visual_lib/src/cuda_common.cpp")
list(REMOVE_ITEM SOURCES ${EXCLUDE_SOURCES})
```

Reply:
not done because I didn't know of this handy feature :)
done.